aws/elb_test.go

```diff
@@ -0,0 +1,89 @@
+package aws
+
+import (
+	"testing"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTestELB(t *testing.T, session *session.Session, name string) {
+	svc := elb.New(session)
+
+	param := &elb.CreateLoadBalancerInput{
+		AvailabilityZones: []*string{
+			awsgo.String("us-west-2a"),
+		},
+		LoadBalancerName: awsgo.String(name),
+		Listeners: []*elb.Listener{
+			&elb.Listener{
+				InstancePort:     awsgo.Int64(80),
+				LoadBalancerPort: awsgo.Int64(80),
+				Protocol:         awsgo.String("HTTP"),
+			},
+		},
+	}
+
+	_, err := svc.CreateLoadBalancer(param)
+	if err != nil {
+		assert.Failf(t, "Could not create test ELB: %s", errors.WithStackTrace(err).Error())
+	}
+}
+
+func TestListELBs(t *testing.T) {
+	session, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String("us-west-2")},
+	)
+
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+
+	elbName := "aws-nuke-test-" + uniqueID()
+	createTestELB(t, session, elbName)
+
+	elbNames, err := getAllElbInstances(session, "us-west-2")
+	if err != nil {
+		assert.Fail(t, "Unable to fetch list of Auto Scaling Groups")
+	}
+
+	assert.Contains(t, awsgo.StringValueSlice(elbNames), elbName)
+}
+
+func TestNukeELBs(t *testing.T) {
```

Comment:
How long do the tests take to run? It seems like it would take a while.

You should probably add:

```go
t.Parallel()
```

As the first line of each test. This will get them all to run in parallel. It will also flush out any concurrency bugs.

Reply:
Tests now run and pass concurrently